From 69a41c776fffe608d0a7708ca6c1d33532c804b0 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Wed, 01 May 2019 11:36:19 +0300
Subject: [PATCH] [SW-0] pcie: replace get_skip_pcie_init_mask() with hl_skip_pcie_init()

This way one can better control PCIe skip.
Use case for this will be the ONIX platform

Change-Id: I1d18a5c397fd403a93cddf43c2e5369df645ff96
---

diff --git a/common/preboot.c b/common/preboot.c
index b2d54b0..21cbeb4 100644
--- a/common/preboot.c
+++ b/common/preboot.c
@@ -89,9 +89,7 @@
 
 static void preboot_pcie_init(void)
 {
-	u32 skip_mask = get_skip_pcie_init_mask();
-
-	if (regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & skip_mask) {
+	if (hl_skip_pcie_init()) {
 		preboot_puts("PCIe initialization is skipped!\n");
 		return;
 	}
diff --git a/common/preboot.h b/common/preboot.h
index b041bde..787deea 100644
--- a/common/preboot.h
+++ b/common/preboot.h
@@ -49,7 +49,7 @@
 
 u64 get_cpu_boot_status_reg(void);
 u64 get_upd_pending_sts_reg(void);
-u32 get_skip_pcie_init_mask(void);
+u32 hl_skip_pcie_init(void);
 void hl_uart_init(void);
 u32 hl_spi_baudr_get(void);
 
diff --git a/gaudi/gaudi.c b/gaudi/gaudi.c
index 47ec761..6bfa80b 100644
--- a/gaudi/gaudi.c
+++ b/gaudi/gaudi.c
@@ -44,10 +44,10 @@
 	return mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_3;
 }
 
-u32 get_skip_pcie_init_mask(void)
+u32 hl_skip_pcie_init(void)
 {
 	/* Use DumpSel[11] to skip PCIe initialization */
-	return BIT(19);
+	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & BIT(19);
 }
 
 u64 hl_pll_get_cpu_pll_base_off(void)
diff --git a/goya/goya.c b/goya/goya.c
index af4e41f..6633585 100644
--- a/goya/goya.c
+++ b/goya/goya.c
@@ -171,10 +171,10 @@
 	return mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3;
 }
 
-u32 get_skip_pcie_init_mask(void)
+u32 hl_skip_pcie_init(void)
 {
 	/* Use DumpSel[8] to skip PCIe initialization */
-	return BIT(21);
+	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & BIT(21);
 }
 
 u64 hl_pll_get_cpu_pll_base_off(void)
